### ISSUE TITLE
xrootd manager crash shortly after restart when cmsd already running

### ISSUE THREAD

**Issue by smithdh** [xrootd/xrootd#1421]
Hello,

On one of our xrootd (5.1.0) managers, if the cmsd is already running it is quite often seen that the xrootd crashes moments after a restart.

```
(gdb) where
#0  XrdNet::Trim (this=0x0, hname=hname@entry=0x7f8d1db9a5a0 "xcache-redirector.cern.ch")
    at /data/dhsmith/xrootd4/xrootd-510-rebuild/src/XrdNet/XrdNet.cc:348
#1  0x00007f8d248df2b2 in XrdLinkCtl::Alloc (peer=..., opts=opts@entry=0)
    at /data/dhsmith/xrootd4/xrootd-510-rebuild/src/Xrd/XrdLinkCtl.cc:158
#2  0x00007f8d248ddf81 in XrdInet::Connect (this=0x7f8d1f8a4150, host=<optimized out>, port=1213, 
    opts=opts@entry=0, tmo=tmo@entry=-1) at /data/dhsmith/xrootd4/xrootd-510-rebuild/src/Xrd/XrdInet.cc:198
#3  0x00007f8d24bd7043 in XrdCmsClientMan::Hookup (this=this@entry=0x7f8d1fb45000)
    at /data/dhsmith/xrootd4/xrootd-510-rebuild/src/XrdCms/XrdCmsClientMan.cc:347
#4  0x00007f8d24bd7880 in XrdCmsClientMan::Start (this=0x7f8d1fb45000)
    at /data/dhsmith/xrootd4/xrootd-510-rebuild/src/XrdCms/XrdCmsClientMan.cc:230
#5  0x00007f8d248a5947 in XrdSysThread_Xeq (myargs=0x7f8d1f874760)
    at /data/dhsmith/xrootd4/xrootd-510-rebuild/src/XrdSys/XrdSysPthread.cc:86
#6  0x00007f8d24451ea5 in start_thread () from /lib64/libpthread.so.0
#7  0x00007f8d237529fd in clone () from /lib64/libc.so.6
```

last few lines in xrootd.log:

```
Config warning: HTTPS functionality was not configured.
210307 16:03:24 29835 sysConfig: XRDROLE:  manager
210307 16:03:24 29835 sysConfig: Configured as HTTP(s) redirector.
------ HTTP protocol initialization completed.
------ xrootd xcache@xcache-redirector.cern.ch:1094 initialization completed.
```

Possibly XrdGlobal::XrdNetTCP is not set in time for the XrdCmsStartManager thread.

**Comment by abh3** [comment]
Hi David,

This is the first we've seen of this. Is it possible to get a stack trace 
with statement numbers?

Andy

**Comment by smithdh** [comment]
Hi Andy,

Here's more:

```
Core was generated by `/usr/bin/xrootd -l /var/log/xrootd/xrootd.log -c /etc/xrootd/xrootd-xcache.cfg'.
Program terminated with signal 11, Segmentation fault.
#0  XrdNet::Trim (this=0x0, hname=hname@entry=0x7f1d17d995a0 "xcache-redirector.cern.ch")
    at /data/dhsmith/xrootd4/xrootd-510-rebuild/src/XrdNet/XrdNet.cc:348
348	  if (Domlen && k > Domlen)
Missing separate debuginfos, use: debuginfo-install xrootd-server-5.1.0-2.el7.x86_64
(gdb) list
343	void XrdNet::Trim(char *hname)
344	{
345	  int k = strlen(hname);
346	  char *hnp;
347	
348	  if (Domlen && k > Domlen)
349	     {hnp = hname + (k - Domlen);
350	      if (!strcmp(Domain, hnp)) *hnp = '\0';
351	     }
352	}
(gdb) print Domlen
Cannot access memory at address 0x20
(gdb) print Domain
Cannot access memory at address 0x18
(gdb) print this
$1 = (XrdNet * const) 0x0
```

**Comment by smithdh** [comment]
Hi Andy. I believe this is fixed by your commit b0642ac. So I'll close this ticket now. Thank you.

**Comment by abh3** [comment]
Yes, it took another site to finally connect the dots. There is a race condition but it's heavily affected by what the server has to load , the order in which it loads it, the configuration order, etc. etc. etc. (let's just say it's timing dependent). Most sites never see this problem. In the end it was a simple fix once I understood the interactions. Thanks for catching this and closing the issue!
